File: check-data.js
#!/usr/bin/env node

const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, 'webapp/lambda/.env.local') });

const { query } = require('./webapp/lambda/utils/database');

async function checkData() {
  console.log('\n=== DATABASE DATA CHECK ===\n');

  const tables = [
    { name: 'stock_symbols', query: 'SELECT COUNT(*) as count FROM stock_symbols' },
    { name: 'price_daily', query: 'SELECT COUNT(*) as count FROM price_daily' },
    { name: 'buy_sell_daily', query: 'SELECT COUNT(*) as count FROM buy_sell_daily' },
    { name: 'buy_sell_weekly', query: 'SELECT COUNT(*) as count FROM buy_sell_weekly' },
    { name: 'buy_sell_monthly', query: 'SELECT COUNT(*) as count FROM buy_sell_monthly' },
    { name: 'technical_data_daily', query: 'SELECT COUNT(*) as count FROM technical_data_daily' },
    { name: 'earnings_estimates', query: 'SELECT COUNT(*) as count FROM earnings_estimates' },
  ];

  for (const table of tables) {
    try {
      const result = await query(table.query);
      const count = parseInt(result.rows[0]?.count || 0);
      const status = count > 0 ? '✅' : '⚠️';
      console.log(`${status} ${table.name}: ${count.toLocaleString()} records`);
    } catch (err) {
      console.log(`❌ ${table.name}: Error - ${err.message.substring(0, 50)}`);
    }
  }

  // Check sample signals
  console.log('\n=== SAMPLE SIGNALS ===\n');
  try {
    const result = await query(`
      SELECT symbol, signal, date, timeframe
      FROM buy_sell_daily
      WHERE signal IN ('Buy', 'Sell')
      LIMIT 5
    `);
    if (result.rows.length > 0) {
      console.log('✅ Sample signals found:');
      result.rows.forEach(row => {
        console.log(`  ${row.symbol} - ${row.signal} on ${row.date}`);
      });
    } else {
      console.log('⚠️ No signals found in buy_sell_daily table');
    }
  } catch (err) {
    console.log(`❌ Error querying signals: ${err.message}`);
  }

  console.log('\n');
  process.exit(0);
}

checkData().catch(err => {
  console.error('Fatal error:', err);
  process.exit(1);
});

File: check_data.py
#!/usr/bin/env python3
import psycopg2
import os

conn = psycopg2.connect(
    host=os.getenv("DB_HOST"),
    port=os.getenv("DB_PORT"),
    database=os.getenv("DB_NAME"),
    user=os.getenv("DB_USER"),
    password=os.getenv("DB_PASSWORD")
)
cur = conn.cursor()

tables = [
    ('stock_scores', 'Swing trader scores'),
    ('buy_sell_daily', 'Buy/sell signals'),
    ('market_health_daily', 'Market health'),
    ('sector_ranking', 'Sector ranking'),
    ('price_daily', 'Price data'),
]

print("Checking data freshness for API endpoints...\n")
for table, desc in tables:
    try:
        cur.execute(f"SELECT COUNT(*) FROM {table}")
        count = cur.fetchone()[0]
        print(f"  {table}: {count:,} rows - {desc}")
    except Exception as e:
        print(f"  {table}: ERROR - {str(e)[:50]}")

# Close failed transaction
cur.close()
conn.rollback()
cur = conn.cursor()

print("\nChecking /api/scores endpoint data...")
try:
    cur.execute("SELECT COUNT(*) FROM stock_scores")
    print(f"  stock_scores: {cur.fetchone()[0]} rows")
except:
    # Try to get schema
    cur.execute("SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'stock_scores'")
    cols = cur.fetchall()
    if cols:
        print("  stock_scores columns: " + ", ".join(f"{c[0]}({c[1]})" for c in cols[:5]))

cur.close()
conn.close()
